### ISSUE TITLE
linter: no-unused-private-class-members reported even though it's used in an expression

### ISSUE THREAD

**Issue by linkej-autodesk** [oxc-project/oxc#19989]
### What version of Oxlint are you using?

1.51.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

no config

### What happened?

If a member `#flag` is used in an expression like `this.#flag && console.log('spam');`, that correctly reports `no-unused-expressions` but also incorrectly reports `no-unused-private-class-members` on `#flag`.

If we fixed `no-unused-expressions`,  the `no-unused-private-class-members` would also disappear, but we're disabling `no-unused-expressions` since it's an old codebase with ~1k instances of that and we're used to that pattern.

More elaborate demonstration:
```js
class Ok {
    #flag = false; // ok
    foo() {
        if (this.#flag) console.log('spam'); // ok
    }
    baz() {
      const x = false; // ok, does not report no-unused-var
      x && console.log('foo'); // ok, reports no-unused-expressions
    }
}

class Bug {
    #flag = false; // bug: reports no-unused-private-class-members
    foo() {
        this.#flag && console.log('spam'); // ok, reports no-unused-expressions
    }
}

export {Bug, Ok}
```

**Comment by camc314** [comment]
> but we're disabling no-unused-expressions since it's an old codebase with ~1k instances of that and we're used to that pattern.

`no-unused-expression` has an option: `allowShortCircuit` that may remove these false positives:

https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-unused-expressions.html#allowshortcircuit

**Comment by linkej-autodesk** [comment]
Thank you! together with allowTernary, that makes the rule usable. And thanks for the fix!

**Comment by camc314** [comment]
Great! - Out of interest, are you migrating to oxlint at autodesk? (if you're allowed to share) I'm just curious - feel free to DM me on X/discord

**Comment by linkej-autodesk** [comment]
Sadly, this is just my own initiative in my specific team. Also I don't know yet whether I will have enough time to actually roll it out. But if I do, I'll be sure to spread the word to the neighbor teams :)
